### PR TITLE
Update

### PR DESCRIPTION
**Pull request by stefanpenner**
- [x] drop most controllers for components
- [x] port tests
- [x] actions up & bindings down
- [x] basic pod layout
- [x] more cleanup.
- [x] update to latest todoMVC styles

cc @hjdivad

### CODE REVIEW

File: app/components/todos-route/component.js [hunk 1]
```diff
@@ -1,38 +1,35 @@
...
-  }.property('active.[]'),
+  }).readOnly(),
 
-  allAreDone: function (key, value) {
+  allAreDone: computed('@each.isCompleted', function (key, value) {
     if (arguments.length === 2) {
-      this.setEach('isCompleted', value);
-      this.invoke('save');
+      // TODO: use action instead of a 2 way CP.
+      var todos = this.get('active');
```
Comment (felixbuenemann) [inline app/components/todos-route/component.js]: Only selecting active todos breaks clearing completed state for all todos. You probably want something like this:
```javascript
var todos = this.get(value ? 'active' : 'completed');
```

File: app/components/todos-route/component.js [hunk 2]
```diff
@@ -1,38 +1,35 @@
...
+  active: filterBy('todos', 'isCompleted', false),
+
+  inflection: computed('active.[]', function() {
     var active = this.get('active.length');
     return active === 1 ? 'item' : 'items';
-  }.property('active.[]'),
+  }).readOnly(),
 
-  allAreDone: function (key, value) {
+  allAreDone: computed('@each.isCompleted', function (key, value) {
```
Comment (felixbuenemann) [inline app/components/todos-route/component.js]: Binding doesn't work for me, should probably be: `computed('todos.@each.isCompleted', …)`

File: app/components/todos-route/component.js [hunk 3]
```diff
@@ -1,38 +1,35 @@
...
     if (arguments.length === 2) {
-      this.setEach('isCompleted', value);
-      this.invoke('save');
+      // TODO: use action instead of a 2 way CP.
+      var todos = this.get('active');
+      todos.setEach('isCompleted', value);
+      todos.invoke('save');
       return value;
     } else {
       return !isEmpty(this) && this.get('length') === this.get('completed.length');
```
Comment (felixbuenemann) [inline app/components/todos-route/component.js]: `this.get('length')`is `undefined`, should probably be: `this.get('todos.length')`

**Comment by felixbuenemann** [comment]
You can use the bower package **todomvc-app-css** and add it to `Brocfile.js` to get the current styles:
```javascript
app.import('bower_components/todomvc-app-css/index.css');
```
I also needed a tiny css fix (not sure what's causing this):
```css
#clear-completed::after { top: 0 } /* Fix "Clear completed" position */
```
And changed to the CSP in `config/environment.js` to allow data-uris for images:
```javascript
ENV.contentSecurityPolicy: {
  'default-src': "'none'",
  'script-src': "'self'",
  'font-src': "'self'",
  'connect-src': "'self'",
  'img-src': "'self' data:",
  'style-src': "'self'",
  'media-src': "'self'"
};
```
Not sure why data-uris are not allowed in the default policy.

**Comment by stefanpenner** [comment]
@hjdivad got time this weekend to continue this effort? I can likely remote pair if you are down.

**Comment by felixbuenemann** [comment]
The css fix I mentioned above has been integrated into todomvc-app-css 1.0.2.

**Comment by felixbuenemann** [comment]
You might also want to take a look at fixing #9 while you're at it.

**Pull request merged**